redecs_agent/redecs_agent.go

```diff
@@ -158,64 +99,14 @@ func main() {
 		sum += 2
 	}
 
-	startFn := func(event *docker.APIEvents) error {
-		var err error
-		container, err := dockerClient.InspectContainer(event.ID)
-		logErrorAndFail(err)
-		service := getServiceName(container)
-		if service != "" {
-			sum = 1
-			for {
-				if err = reportServiceActive(service); err == nil {
-					break
-				}
-				if sum > 8 {
-					log.Errorf("Error reporting service %s active", service)
-					break
-				}
-				time.Sleep(time.Duration(sum) * time.Second)
-				sum += 2
-			}
-		}
-		log.Debugf("Docker %s started", event.ID)
-		return nil
-	}
-
-	stopFn := func(event *docker.APIEvents) error {
-		var err error
-		container, err := dockerClient.InspectContainer(event.ID)
-		logErrorAndFail(err)
-		service := getServiceName(container)
-		if service != "" {
-			sum = 1
-			for {
-				if err = reportServiceInactive(service); err == nil {
-					break
-				}
-				if sum > 8 {
-					log.Errorf("Error reporting service %s inactive", service)
-					break
-				}
-				time.Sleep(time.Duration(sum) * time.Second)
-				sum += 2
-			}
-		}
-		log.Debugf("Docker %s stopped", event.ID)
-		return nil
-	}
-
-	dockerRouter, err := docker_router.AddStartStopHandlers(startFn, stopFn, dockerClient)
-	defer dockerRouter.Stop()
-	logErrorAndFail(err)
-	dockerRouter.Start()
-
-	log.Debug("Waiting for Docker events")
+	log.Debug("redecs_agent started")
 
 	// continue processing once per minute
-	ticker := time.NewTicker(checkInterval)
+	ticker := time.NewTicker(reportInterval)
+	serviceName := getServiceName()
```

Comment:
Load this at the very beginning, exit with a descriptive error message and non-zero exit code if serviceName the empty string (https://godoc.org/os#Getenv)

- This script is misconfigured (or being used improperly) if there is no service name